Tasks/Common/VstsAzureRestHelpers_/VstsAzureRestHelpers_.psm1

```diff
@@ -377,6 +391,53 @@ function Get-SpnAccessToken {
     }
 }
 
+# Get the Bearer Access Token from the Endpoint
+function Get-MsiAccessToken {
+    [CmdletBinding()]
+    param([Parameter(Mandatory=$true)] $endpoint)
+
+    $msiPort = 50342;
```

Comment:
This logic has to change.

Reply:
We are switching to IMDS service: https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service 

IMDS is now generally available as per the doc link above.

Reply:
Done. updated.